File: backend/albergue1/services/raza.py
from flask import Blueprint, request, jsonify, make_response
from utils.db import db

from models.raza import Raza
from schemas.raza import raza_schema, razas_schema

raza_routes = Blueprint('raza_routes', __name__)

@raza_routes.route("/get_razas", methods=["GET"])
def get_razas():
    razas = Raza.query.all()

    if not razas:
        data = {
            'message': 'No se encontraron razas.',
            'status': 404
        }
        return make_response(jsonify(data), 404)
    else:
        result = razas_schema.dump(razas)
        data = {
            'message': 'Razas encontradas.',
            'status': 200,
            'data': result
        }
        return make_response(jsonify(data), 200)
    
@raza_routes.route("/get_raza", methods=["POST"])
def get_raza():
    id_raza = request.json.get('id_raza')
    if not id_raza:
        data = {
            'message': 'ID de raza no proporcionado.',
            'status': 400
        }
        return make_response(jsonify(data), 400)
    else:
        raza = Raza.query.get(id_raza)

        if not raza:
            data = {
                'message': 'Raza no encontrada.',
                'status': 404
            }
            return make_response(jsonify(data), 404)
        else:
            result = raza_schema.dump(raza)
            data = {
                'message': 'Raza encontrada.',
                'status': 200,
                'data': result
            }
            return make_response(jsonify(data), 200)

File: backend/albergue3/controllers/raza.js
const Raza = require('../models/raza.js');

const get_razas = (req, res) => {
    Raza.obtenerTodos((err, resultados) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (resultados.length === 0) {
            return res.status(404).json({ mensaje: 'No se encontraron razas' });
        }
        res.status(200).json(resultados);
    });
};

const get_raza = (req, res) => {
    const { id_raza } = req.body;
    if (!id_raza) {
        return res.status(400).json({ mensaje: 'ID de raza es requerido' });
    }
    
    Raza.obtenerPorId(id_raza, (err, resultado) => {
        if (err) {
            return res.status(500).json({ mensaje: 'Error en el servidor', error: err });
        }
        if (!resultado || resultado.length === 0) {
            return res.status(404).json({ mensaje: 'Raza no encontrada' });
        }
        res.status(200).json(resultado[0]);
    });
}

module.exports = {
    get_razas,
    get_raza
};
